sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVInferSchema.scala

```diff
@@ -168,6 +169,15 @@ class CSVInferSchema(val options: CSVOptions) extends Serializable {
     // This case infers a custom `dataFormat` is set.
     if ((allCatch opt timestampParser.parse(field)).isDefined) {
       TimestampType
+    } else {
+      tryParseDateFormat(field)
+    }
+  }
+
+  private def tryParseDateFormat(field: String): DataType = {
+    if ((allCatch opt new SimpleDateFormat(
```

Comment:
Should avoid using `SimpleDateFormat` from JDK7. It has multiple flaws such as disallowing exact match. You should use `DateFormatter`.

Reply:
@HyukjinKwon  - Sorry , I forget to push the latest commit  where I was setting setLenient(false) , This will help with the exact match.

```
  val dtFormat = new SimpleDateFormat(options.dateFormat, Locale.US)
      dtFormat.setLenient(false)
      dtFormat.parse(field)}).isDefined) {
      DateType
    }
```